rfcs/0000-upgradeable-lifecycle.md

```diff
@@ -0,0 +1,82 @@
+**Problem:**
+Currently, the lifecycle version used in kpack is not configurable, nor is it user upgradeable. Users must wait for a new kpack release to be able to upgrade the version of the lifecycle used by their builders.
+
+**Outcome:**
+We would like to decouple the lifecycle upgrade process from the kpack upgrade process, allowing users to upgrade the lifecycle on their own.
+
+**Actions to take:**
```

Comment:
I think it would be useful to discuss how the lifecycle is monolithically configured. Currently, it is provided as a an env var on the kpack controller so, updating that would require a kpack controller restart. Perhaps, a monitored configmap would make sense?